src/planner/binder/statement/bind_create_table.cpp

```diff
@@ -168,11 +319,48 @@ void Binder::BindDefaultValues(vector<ColumnDefinition> &columns, vector<unique_
 	}
 }
 
+void ReplaceGeneratedColumnReferences(unique_ptr<ParsedExpression> &expr, const BoundCreateTableInfo &table) {
+	if (expr->type == ExpressionType::COLUMN_REF) {
+		auto &column_ref = (ColumnRefExpression &)*expr;
+		auto &name = column_ref.GetColumnName();
+		auto entry = table.name_map.find(name);
+		if (entry == table.name_map.end()) {
+			return;
+		}
+		if (entry->second.column_type != TableColumnType::GENERATED) {
+			return;
+		}
+		auto index = entry->second.index;
+		auto &base = (CreateTableInfo &)*table.base;
+		auto &generated_column = base.columns[index];
+		expr = generated_column.GeneratedExpression().Copy();
+		ReplaceGeneratedColumnReferences(expr, table);
+		return;
+	}
+	ParsedExpressionIterator::EnumerateChildren(
+	    *expr, [&](unique_ptr<ParsedExpression> &child) { ReplaceGeneratedColumnReferences(child, table); });
+}
+
+void CreateGeneratedCheckConstraints(BoundCreateTableInfo &info) {
+	auto &base = (CreateTableInfo &)*info.base;
+	for (idx_t i = 0; i < base.columns.size(); i++) {
+		auto &col = base.columns[i];
+		if (!col.Generated()) {
+			continue;
+		}
+		auto expression = col.GeneratedExpression().Copy();
+		ReplaceGeneratedColumnReferences(expression, info);
+		base.constraints.push_back(make_unique<GeneratedCheckConstraint>(i, move(expression)));
```

Comment:
Maybe it is easier not to use check constraints but to directly verify the generated columns in the `data_table.cpp` (`VerifyAppendConstraints`) since then we can also throw more clear errors and won't need to work around check constraint behavior